### PR TITLE
add test infrastructure

### PR DESCRIPTION
**Pull request by Boshen**

### CODE REVIEW

File: src/docs/learn/architecture/test.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+To ensure that the parser does not panic when encountering random data, three fuzzers are used:
+
+1. [cargo fuzz](https://github.com/rust-fuzz/cargo-fuzz) for [sending random bytes](https://github.com/oxc-project/oxc-fuzz-parser/blob/main/fuzz/fuzz_targets/parser.rs) to the parser.
+2. [shift-fuzzer-js](https://github.com/shapesecurity/shift-fuzzer-js) by [bakkot](https://github.com/bakkot) for producing random by valid ASTs.
+3. [Automated-Fuzzer](https://github.com/qarmin/Automated-Fuzzer) by [qarmin](https://github.com/qarmin), who [actively reports](https://github.com/oxc-project/oxc/issues?q=is%3Aissue+author%3Aqarmin+) crashes.
+
+### Conformance
+
+All parser tests from [Test262](https://github.com/tc39/test262), [Babel](https://github.com/babel/babel), and [TypeScript](https://github.com/microsoft/TypeScript) are used to test JavaScript, TypeScript, and JSX syntax.
```
Comment (DonIsaac) [inline src/docs/learn/architecture/test.md:33]: We also have the scope tests from Ts-eslint for Semantic

File: src/docs/learn/architecture/test.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+We spend half of our time on strengthening the test infrastructure to prevent problems from propagating to downstream tools.
+
+## Parser
+
+### Memory Safety
+
+The oxc parser uses [`bumpalo`](https://docs.rs/bumpalo/latest/bumpalo) as the memory allocator for its AST.
+None of the AST nodes have a `drop` implementation.
+Miri [is used](https://github.com/oxc-project/oxc/actions/workflows/miri.yml) to ensure that no heap-allocated data is stored on the AST nodes
+so that memory is leaked when the memory allocator is dropped.
```
Comment (DonIsaac) [inline src/docs/learn/architecture/test.md:21]: Should we talk about memory safety in the lexer and SourceCode? Those use a lot of unsafe code.

File: src/docs/learn/architecture/test.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+2. [shift-fuzzer-js](https://github.com/shapesecurity/shift-fuzzer-js) by [bakkot](https://github.com/bakkot) for producing random by valid ASTs.
+3. [Automated-Fuzzer](https://github.com/qarmin/Automated-Fuzzer) by [qarmin](https://github.com/qarmin), who [actively reports](https://github.com/oxc-project/oxc/issues?q=is%3Aissue+author%3Aqarmin+) crashes.
+
+### Conformance
+
+All parser tests from [Test262](https://github.com/tc39/test262), [Babel](https://github.com/babel/babel), and [TypeScript](https://github.com/microsoft/TypeScript) are used to test JavaScript, TypeScript, and JSX syntax.
+
+For test262, all stage 4 and regular expression tests are included.
+
+All conformance results are stored in a snapshot file for tracking changes:
```
Comment (DonIsaac) [inline src/docs/learn/architecture/test.md:37]: We should mention that every single tool has its own conformance tests, not just the parser

File: src/docs/learn/architecture/test.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
+# Test Infrastructure
+
+::: info
+
+This article serves as an invitation for sharing ideas to improve our test infrastructure,
+feel free to contact me on [twitter](https://x.com/boshen_c) or [discord](https://discord.gg/9uXCAwqQZW).
```
Comment (DonIsaac) [inline src/docs/learn/architecture/test.md:6]: Feel free to throw my name on here if you want. I'm around to help with support.

File: src/docs/learn/architecture/test.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+:::
+
+In Oxc, correctness and reliability are taken extremely seriously.
+
+We spend half of our time on strengthening the test infrastructure to prevent problems from propagating to downstream tools.
```
Comment (DonIsaac) [inline src/docs/learn/architecture/test.md:12]: Should we link to our CodeCov results here?

File: src/docs/learn/architecture/test.md (hunk 2 above)
Comment (Boshen) [reply src/docs/learn/architecture/test.md]: let's not mention `unsafe`.

File: src/docs/learn/architecture/test.md (hunk 4 above)
Comment (Boshen) [reply src/docs/learn/architecture/test.md]: changed to only discord.

File: src/docs/learn/architecture/test.md (hunk 1 above)
Comment (Boshen) [reply src/docs/learn/architecture/test.md:18]: I'll leave it out.

**Review by DonIsaac — APPROVED** [review]

**Comment by Boshen** [comment]
I'll merge and let @overlookmotel take a look at this when he has time.

**Pull request merged**